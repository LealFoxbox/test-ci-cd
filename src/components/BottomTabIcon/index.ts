export * from './BottomTabIcon';
export { default } from './BottomTabIcon';
