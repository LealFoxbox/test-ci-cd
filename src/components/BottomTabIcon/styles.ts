import styled from 'styled-components/native';

export const iconSize = 28;
const lineExtraWidth = 12;

export const Container = styled.View`
  flex: 1;
`;

export const IconContainer = styled.View`
  flex: 1;
  justify-content: center;
  margin-left: ${lineExtraWidth}px;
`;

export const Line = styled.View<{ focused: boolean; color?: string }>`
  width: ${iconSize + lineExtraWidth * 2}px;
  height: 2px;
  background-color: ${({ focused, color }) => (focused && color ? color : 'transparent')};
`;
