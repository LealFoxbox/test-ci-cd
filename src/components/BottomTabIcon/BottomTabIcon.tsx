import React from 'react';
import { IconProps } from 'react-native-vector-icons/Icon';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import { Text } from 'react-native-paper';

import { Container, IconContainer, iconSize } from './styles';

export interface BottomTabIconProps extends IconProps {
  icon: string;
}

const BottomTabIcon: React.FC<BottomTabIconProps> = ({ icon, name, ...props }) => {
  return (
    <Container>
      <IconContainer>
        <MaterialIcons {...props} name={icon} size={iconSize} />
      </IconContainer>
      <Text>{name}</Text>
    </Container>
  );
};

export default BottomTabIcon;
