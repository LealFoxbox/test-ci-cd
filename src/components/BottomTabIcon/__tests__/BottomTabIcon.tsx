import React from 'react';
import { render } from '@testing-library/react-native';

import BottomTabIcon from '../BottomTabIcon';

describe('BottomTabIcon', () => {
  it('should render', () => {
    const result = render(<BottomTabIcon icon="warning" name="warning" />);

    expect(result).toMatchSnapshot();
  });
});
