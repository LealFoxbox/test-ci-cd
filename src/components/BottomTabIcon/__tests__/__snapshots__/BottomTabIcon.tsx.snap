// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomTabIcon should render 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "marginLeft": 12,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 28,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      î€‚
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        undefined,
      ]
    }
  >
    warning
  </Text>
</View>
`;
